opcUaDevSupApp/drvOpcUa.cpp

```diff
@@ -801,7 +801,9 @@ long OpcWriteValue(int opcUaItemIndex,double val,int verbose)
     return 0;
 }
 /* iocShell: record write func  */
+extern "C" {
```

Comment:
OK, clarified.
`epicsExport.h` uses `extern "C"{}`, but only defines macros.
When you *use* the macros, those statements do not apply, and the variable you declare is getting a mangled name. GCC doesn't mangle the name anyway, so linking works.

Your solution to add the `extern "C" {}` around every use of the macro is correct.